plugins/manager/fragments/yform/manager/history.diff.php

```diff
@@ -0,0 +1,205 @@
+    }
+
+foreach ($data as $field => $value) {
+    ++$diffs['removed']['count'];
+    $diffs['removed']['rows'] .= '
+        <tr>
+            <th><i data-toggle="tooltip" data-placement="top" title="" class="rex-icon fas fa-question" data-original-title="' .
+                rex_i18n::msg('yform_manager_type_name') . ': ' . rex_i18n::msg('yform_manager_type_unknown') . '"></i> ' . $field . '</th>
+            <td><em>' . rex_i18n::msg('yform_history_diff_no_longer_existing') . '</em></td>
+            <td>' . $value . '</td>
+        </tr>';
+}
+
+// build restore url
+$restoreUrl = http_build_query([
+    'table_name' => $table->getTableName(),
+    'func' => 'history',
+    'subfunc' => 'restore',
+    'data_id' => $datasetId,
+    'history_id' => $historyId,
+]) . http_build_query(rex_csrf_token::factory($this->getVar('csrf_key', ''))->getUrlParams());
+
+$content = '
+    <div class="modal-header">
+        <button type="button" class="close" data-dismiss="modal" aria-hidden="true">&times;</button>
+        <h4 class="modal-title">
+            ' . rex_i18n::msg('yform_history_dataset_id') . ': ' . $datasetId . '
+            <small>[' . date('d.m.Y H:i:s', strtotime($timestamp)) . ']</small>
+        </h4>
+    </div>
+    <div class="modal-body panel-default">
+';
+
+foreach ($diffs as $change => $diff) {
+    $content .= '
+        <header class="panel-heading" ' . ('' !== $diff['rows'] ? 'data-toggle="collapse" data-target="#collapse-history-table-' . $change . '"' : '') . '>
+            <div class="panel-title"><i class="rex-icon ' . $diff['icon'] . '"></i> ' . rex_i18n::msg('yform_history_diff_headline_' . $change) .
+                ' [' . ($diff['count'] > 0 ? '<b>' : '') . $diff['count'] . ($diff['count'] > 0 ? '</b>' : '') . ']' . '
+            </div>
+        </header>
+        <div id="collapse-history-table-' . $change . '" ' . ('' !== $diff['rows'] ? 'class="panel-collapse collapse ' . ('changed' === $change ? 'in' : '') . '"' : '') . '>';
+
+    if('' !== $diff['rows']) {
+        $content .= '
+            <table class="table history-diff-table" data-change-mode="' . $change . '">
+                <thead>
+                    <tr>
+                        <th class="rex-table-width-7 field-name"><i class="rex-icon-"></i>' . rex_i18n::msg('yform_tablefield') . '</th>
+                        <th class="rex-table-width-10">' . rex_i18n::msg('yform_history_dataset_current') . '</th>
+                        <th class="rex-table-width-10">' . date('d.m.Y H:i:s', strtotime($timestamp)) . '</th>
+                    </tr>
+                </thead>
+                <tbody>' . $diff['rows'] . '</tbody>
+            </table>';
+    }
+
+    $content .= '</div>';
+}
+
+$content .= '
+    </div>
+    <div class="modal-footer">
+        <a href="index.php?page=yform/manager/data_edit?' . $restoreUrl . '" class="btn btn-warning" onclick="return confirm(\'' . rex_i18n::msg('yform_history_restore_confirm') . '\')">' .
```

Comment:
kein onclick bitte, sondern js in Dateien auslagern. CSP macht da sonst Probleme

Reply:
wird gemacht